File: Mobiles/flutter/lib/models/restrictions.dart
class Restrictions {
  int maxCaloriesPerSlice;
  bool mustBeVegetarian;
  List<String> excludedIngredients;
  List<String> excludedTools;
  int maxNumberOfToppings;
  int minNumberOfToppings;
  String customName;

  Restrictions({
    this.maxCaloriesPerSlice = 1000,
    this.mustBeVegetarian = false,
    this.excludedIngredients = const [],
    this.excludedTools = const [],
    this.maxNumberOfToppings = 5,
    this.minNumberOfToppings = 2,
    this.customName = '',
  });

  Map<String, dynamic> toJson() {
    return {
      'maxCaloriesPerSlice': maxCaloriesPerSlice,
      'mustBeVegetarian': mustBeVegetarian,
      'excludedIngredients': excludedIngredients,
      'excludedTools': excludedTools,
      'maxNumberOfToppings': maxNumberOfToppings,
      'minNumberOfToppings': minNumberOfToppings,
      'customName': customName,
    };
  }
}

File: Mobiles/ios/QuickPizzaIos/Features/Pizza/Models/Restrictions.swift
import Foundation

/// Preferences for pizza recommendations.
struct Restrictions: Codable, Equatable {
    var maxCaloriesPerSlice: Int
    var mustBeVegetarian: Bool
    var excludedIngredients: [String]
    var excludedTools: [String]
    var maxNumberOfToppings: Int
    var minNumberOfToppings: Int
    var customName: String

    static let `default` = Restrictions(
        maxCaloriesPerSlice: 1000,
        mustBeVegetarian: false,
        excludedIngredients: [],
        excludedTools: [],
        maxNumberOfToppings: 5,
        minNumberOfToppings: 2,
        customName: ""
    )
}
